### DISCUSSION TITLE
How to get all components that implement a specific trait

### DISCUSSION THREAD

**Discussion by slangerosuna** [bevyengine/bevy#10096] [Q&A] [answered]
I'm making a multiplayer plugin, and for one of the systems, I need to access all of the components that implement this 'Serializable' trait
```rs
pub trait Serializable {
      fn from_bytes(&mut self, bytes: &[u8]);
      fn to_bytes(&self) -> Vec<u8>;

      fn get_type_id(&self) -> u16;
}
```
How would I go about making a query that gets all components that implement this trait?
I am aware that I can make it a super trait of component which would allow me to query it, but that only gives me one component that implements this trait, instead of all of them, but I am unaware as to how I could get all of these components preferably in the form of an Iterator.

**Comment by alice-i-cecile** [comment, marked as answer]
You want https://crates.io/crates/bevy-trait-query :) Eventually we may upstream it, but for now it's well-maintained and popular.

**Reply by slangerosuna** [reply]
thank you very much, that is definitely what I wanted
